### PR TITLE
fix(grainfmt): Indent function application args when adding parens

### LINKED ISSUE #1087: Grainfmt: Evaluate a couple of oddities in stdlib formatting
Reference the PR for the couple of code chunks I commented on. They seem to deal with wrapping when code is inside parens—we might want to indent a bit.

_Originally posted by @phated in https://github.com/grain-lang/grain/pull/1084#discussion_r777047772_

### PR DESCRIPTION
**Pull request by marcusroberts**
Fixes #1087 

Add indenting when adding parens around function application arguments for better layout.

Added tests and reformatted the stdlib with the change

### CODE REVIEW

File: stdlib/string.gr [hunk 1]
```diff
@@ -1635,7 +1635,7 @@ let rec decodeRangeHelp =
             // high surrogate. next character is low srurrogate
             let w1 = (w1 & 0x03FFn) << 10n
             let w2 = (WasmI32.load8U(bytesPtr, 2n) << 8n |
-              WasmI32.load8U(bytesPtr, 3n)) &
+                WasmI32.load8U(bytesPtr, 3n)) &
```
Comment (phated) [inline stdlib/string.gr:1638]: This seems wrong

Comment (marcusroberts) [reply stdlib/string.gr:1638]: It does.

Comment (marcusroberts) [reply stdlib/string.gr:1638]: You might want
(
  WasmI32.load8U(bytesPtr, 2n) << 8n | 
  WasmI32.load8U(bytesPtr, 3n)
 ) &
0x03FFn

to make that clearer.  I can make it do that easily if that's more comprehensible (which I think it is)

Comment (marcusroberts) [reply stdlib/string.gr:1638]: (imagine the two WasmI32 are indented, the comment isn't showing it)

Comment (marcusroberts) [reply stdlib/string.gr]: I've made that code change and I think comprehension has jumped up a lot

**Review by phated — APPROVED** [review]
The results look good to me. Only comment about implementation is that it is unfortunate that we are still reliant on `willBreak` but 🤷 

Wondering what @ospencer thinks about the formatting results.

File: compiler/test/formatter_outputs/application_indenting.gr [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+}
+
+let grisuRound = (buffer, len, delta, rest, ten_kappa, wp_w) => {
+  let mut lastp = buffer + len - 1n
+  let mut digit = Module32.load8U(lastp, 0n)
+  let mut rest = rest
+  while (
+    Module64.ltU(rest, wp_w) &&
+    Module64.geU(Module64.sub(delta, rest), ten_kappa) &&
+    (
```
Comment (ospencer) [inline compiler/test/formatter_outputs/application_indenting.gr:17]: I don't _love_ this. Here's what Prettier did though: https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAbOMAEBzATgSwGcBXAJQmKgBNMBeTACgCNiAzVuXAGk3Sh6pxUMAIY9ccQjB7woAfQDWIgA7KxmAO7K5GgJR0AfJmAAdKJl4ZeIqcrqYW7TpgDUl8wFpMARihmL6FhU+Nj4WPQAshBUxOgAzABMAHSoECJUABwAqgyoNjDKPAAMULr+llgSUvZVMOUaABb46IzlFlEx6ABsACwpMDm1PFo6+gBkY22YHbFwvUnYcDkz3X0kTAyCwuq1ujIIiipq45PmFq1n59PRs-PCyzerSelUDEOYsoeqInua2nqYAA+gKm7Uecz62AGDFB5xWEKS6wYIw04kkMD2sLBnQRSPh8xeb3R+3kSm+vxRZUu5ypV1pmH0pmpwVCWA89F8U1q9nxfUJ70+ZOO5QAvuV4YlETAIBIMrl8oVMCywsVSmYRSAuCAIMoYPhoIRkKARLhcBANAAFE0IQ0oESoDQiACehq1TFwIjACgwAGURABbOAAGXwUDgyFY9sIcDdHq9vrUYFD2GQMFwxBjIDg-qYcCogioQZEUGwxBEiwAYjL-SIYHqS8gQCJiNLNSAGjB-agAOpNeCERNwH02sL4ABuYSdjbAhFdIFD0dwMAtHuwNYjUczACtCAAPH3J9AARWIEHgG9Q0a1alwi8bTBEudQbeUBFg3fwVBgDWQGSK17NaNuw9ZRG1fSRODHcMtQAR1PeAVx1W0m0IDwwzzPM2wkOD8AkFdy3XJBI0vTNo39fBU3TUjDzgE8z3DIjNy1UQmA-L8fyQBJmI9ZpkwAYQgf1CKzQgAFY22IaMABVH1tYirxAMcMwASWoBAYB9MACF1ABBagfRgJ10AvaMRRFIA

Comment (marcusroberts) [reply compiler/test/formatter_outputs/application_indenting.gr]: I've made a tweak, it now matches prettier, let me know what you think about the other changes this produces.   I think they look OK

**Review by ospencer — APPROVED** [review]
Nice!

**Pull request merged**